### PR TITLE
[CI] When the CI is starting the server (chip-tool or darwin-framework-tool) wait to see for the websocket message ready before trying to connect

### PR DESCRIPTION
**Pull request by vivien-apple**
#### Problem

When CI is running YAML with `chip-tool` or `Darwin-framework-tool` it sometimes fail to connect properly to the runner. This PR allow a maximum 30 seconds for the web socket server to start before trying to connect. If the web socket starts before, we will try to connect as soon as possible, otherwise we would just raise an error after `30` seconds.

### CODE REVIEW

**Comment by vivien-apple** [comment]
One such occurence of an issue to connect is https://github.com/project-chip/connectedhomeip/actions/runs/7817637999/job/21326019576 and we really don't know why...

**Review by yunhanw-google — APPROVED** [review]

**Comment by vivien-apple** [comment]
I have updated the code such that the maximum number of retries is set back to `3` (so it basically reverts https://github.com/project-chip/connectedhomeip/commit/db067cb69251254cf276c293f15dfeda909106e9).
Also I wonder if some of the connection failures comes from the port not being released properly. For the sake of checking I have updated the code to kill the process with `process.termiante()` instead of always doing `process.kill()`.

File: scripts/py_matter_yamltests/matter_yamltests/websocket_runner.py [hunk 1]
```diff
@@ -24,6 +27,10 @@
 
 _KEEP_ALIVE_TIMEOUT_IN_SECONDS = 120
 _MAX_MESSAGE_SIZE_IN_BYTES = 10485760  # 10 MB
+_CONNECT_MAX_RETRIES_DEFAULT = 3
```
Comment (bzbarsky-apple) [inline scripts/py_matter_yamltests/matter_yamltests/websocket_runner.py:30]: This used to be 4 before https://github.com/project-chip/connectedhomeip/commit/db067cb69251254cf276c293f15dfeda909106e9, no?

File: scripts/py_matter_yamltests/matter_yamltests/websocket_runner.py [hunk 2]
```diff
@@ -93,15 +100,48 @@ async def _stop_client(self, instance):
...
+            # Loop to read the subprocess output with a timeout
+            lines = []
+            while True:
+                if time.time() - start_time > _WEBSOCKET_SERVER_MESSAGE_TIMEOUT:
+                    for line in lines:
+                        print(line, end='')
+                    self._hooks.abort(url)
+                    instance.kill()
+                    raise Exception(
+                        f'Connecting to {url} failed. The websocket startup has not been detected.')
```
Comment (bzbarsky-apple) [inline scripts/py_matter_yamltests/matter_yamltests/websocket_runner.py:121]:
```suggestion
                        f'Connecting to {url} failed. Websocket startup has not been detected.')
```

File: scripts/py_matter_yamltests/matter_yamltests/websocket_runner.py [hunk 3]
```diff
@@ -93,15 +100,48 @@ async def _stop_client(self, instance):
...
+                command, text=True, bufsize=1, stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
+
+            # Loop to read the subprocess output with a timeout
+            lines = []
+            while True:
+                if time.time() - start_time > _WEBSOCKET_SERVER_MESSAGE_TIMEOUT:
+                    for line in lines:
+                        print(line, end='')
+                    self._hooks.abort(url)
+                    instance.kill()
```
Comment (bzbarsky-apple) [inline scripts/py_matter_yamltests/matter_yamltests/websocket_runner.py:119]: Should this also do the "terminate, then kill" thing?  Might want that in a helper method....

**Review by bzbarsky-apple — APPROVED** [review]

File: scripts/py_matter_yamltests/matter_yamltests/websocket_runner.py (hunk 3 above)
Comment (vivien-apple) [reply scripts/py_matter_yamltests/matter_yamltests/websocket_runner.py]: I used `self._stop_server`

File: scripts/py_matter_yamltests/matter_yamltests/websocket_runner.py (hunk 1 above)
Comment (vivien-apple) [reply scripts/py_matter_yamltests/matter_yamltests/websocket_runner.py]: Indeed.

**Pull request merged**